### DISCUSSION TITLE
Why there is "response status_code should be in range [200, 1000), not 100"?

### DISCUSSION THREAD

**Discussion by aamibhoot** [fastapi/fastapi#10017] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.get("/")
def index():
   raise HTTPException(status_code=status.HTTP_100_CONTINUE, detail="Cheez!")
```

### Description

When the endpoint is hits; the app crashes and shows: `Response status_code should be in range [200, 1000), not 100`

1. So FastAPI don't support status code 100?
2. There are maybe some typos like: `1000` and brackets?

### Operating System

macOS

### FastAPI Version

0.100.0

### Pydantic Version

2.0.2

### Python Version

3.11.2

**Comment by jgould22** [comment, marked as answer]
For reference https://www.rfc-editor.org/rfc/rfc9110.html#name-status-codes

100s are interim responses, ie an Http exception which is going to respond and close the connection should not contain 100 status.

The other is not a typo, while > 599 status codes are not technically valid and should be treated as a 500 some places do use that to provide extra information to the client. You can see that mentioned here https://www.rfc-editor.org/rfc/rfc9110.html#section-15-6

Oh and with regards to the brackets that is just a little mathematical interval notation .

When describing ranges in `[]` means inclusive and `()` means exclusive. So in the above message it is telling you that the number can be any number 200 to 999 but not 1000.

**Reply by aamibhoot** [reply]
Wow, that was interesting 🙆🏻‍♂️. Learned something new today. Thanks ❤️
